### ISSUE TITLE
ERROR: No matching distribution found for anaconda-project>=0.9.1

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython-wheels#8]
I just tried added Python 3.9 to the build target, both AppVeyor and Travis are failing with:

```
python -m pip install git+https://github.com/Anaconda-Server/anaconda-client.git
Collecting git+https://github.com/Anaconda-Server/anaconda-client.git
  Cloning https://github.com/Anaconda-Server/anaconda-client.git to c:\users\appveyor\appdata\local\temp\1\pip-req-build-ik1c6f0i
  Running command git clone -q https://github.com/Anaconda-Server/anaconda-client.git 'C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-ik1c6f0i'
Collecting clyent>=1.2.0
  Downloading clyent-1.2.1.tar.gz (20 kB)
Collecting nbformat>=4.4.0
  Downloading nbformat-5.1.3-py3-none-any.whl (178 kB)
Collecting python-dateutil>=2.6.1
  Downloading python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
ERROR: Could not find a version that satisfies the requirement anaconda-project>=0.9.1 (from anaconda-client) (from versions: none)
ERROR: No matching distribution found for anaconda-project>=0.9.1
```

This is the post-build step to push the wheels to https://anaconda.org/multibuild-wheels-staging

**Comment by matthew-brett** [comment]
Do you already have something like this?

https://github.com/MacPython/nipy-wheels/blob/master/.travis.yml#L125

**Comment by peterjc** [comment]
Not yet, I was about to try the ``pip install git+https://github.com/Anaconda-Platform/anaconda-project`` work around after filing https://github.com/Anaconda-Platform/anaconda-client/issues/562

Glad to see I was on the right track, thank you!

**Comment by peterjc** [comment]
Noticed now you had:

```
    - pip install git+https://github.com/Anaconda-Platform/anaconda-project
    - pip install git+https://github.com/Anaconda-Platform/anaconda-client
```

Right now my URLs are slightly different:

```
    - python -m pip install git+https://github.com/Anaconda-Platform/anaconda-project
    - python -m pip install git+https://github.com/Anaconda-Server/anaconda-client
```

Looks like https://github.com/Anaconda-Server/anaconda-client was renamed to https://github.com/Anaconda-Platform/anaconda-client since that gives an automatic redirection now.

**Comment by peterjc** [comment]
CI still running but looks like partial success, Windows Python 3.9 worked which was the primary goal today. About to update the submodules which hopefully will sort most of this.

**Comment by peterjc** [comment]
Windows looks good on AppVeyor, but don't see them on https://anaconda.org/multibuild-wheels-staging/biopython/files?version=1.78

Mac looks good on Travis, see them on https://anaconda.org/multibuild-wheels-staging/biopython/files?version=1.78

Linux trouble still on Travis, all failing at the wheel upload stage

```
$ python -m pip install git+https://github.com/Anaconda-Platform/anaconda-project
Collecting git+https://github.com/Anaconda-Platform/anaconda-project
  Cloning https://github.com/Anaconda-Platform/anaconda-project to /tmp/pip-VF_GYU-build
Collecting anaconda-client (from anaconda-project==0.9.1+95.ga07a185)
/home/travis/build/biopython/biopython-wheels/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/home/travis/build/biopython/biopython-wheels/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Could not find a version that satisfies the requirement anaconda-client (from anaconda-project==0.9.1+95.ga07a185) (from versions: )
No matching distribution found for anaconda-client (from anaconda-project==0.9.1+95.ga07a185)
```

**Comment by peterjc** [comment]
Not sure why only Linux gave this error, but it looks like anaconda-client while requires  anaconda-project>=0.9.1 that in turn requires anaconda-client (circular dependency at the time of writing). So trying installing both at once:

```
pip install git+https://github.com/Anaconda-Platform/anaconda-project git+https://github.com/Anaconda-Platform/anaconda-client
```

https://github.com/biopython/biopython-wheels/commit/b89a1c25e413bce2711647da37e1b8dfcaceb7c3

**Comment by peterjc** [comment]
Looking back to when we did Biopython 1.78, the lack of Windows upload rings a bell - opened #9 on that, but easy workaround with the wheel artefacts collected by AppVeyor.

**Comment by peterjc** [comment]
Bah, going in circles with the anaconda-client - now getting:

```
$ python -m pip install git+https://github.com/Anaconda-Platform/anaconda-project git+https://github.com/Anaconda-Platform/anaconda-client
Collecting git+https://github.com/Anaconda-Platform/anaconda-project
  Cloning https://github.com/Anaconda-Platform/anaconda-project to /tmp/pip-Zz1psx-build
Collecting git+https://github.com/Anaconda-Platform/anaconda-client
  Cloning https://github.com/Anaconda-Platform/anaconda-client to /tmp/pip-JaqyuB-build
Collecting anaconda-client (from anaconda-project==0.9.1+95.ga07a185)
/home/travis/build/biopython/biopython-wheels/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/home/travis/build/biopython/biopython-wheels/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Could not find a version that satisfies the requirement anaconda-client (from anaconda-project==0.9.1+95.ga07a185) (from versions: )
No matching distribution found for anaconda-client (from anaconda-project==0.9.1+95.ga07a185)
```

**Comment by peterjc** [comment]
For some reason the Linux wheel deployment seems to be running under Python 2.7, thus all the security warnings from pip which is probably what breaks it finding the ``anaconda-client`` dependencies. e.g. Even using the most recent prior to the circular dependency, we get:

```
$ python -m pip install git+https://github.com/Anaconda-Platform/anaconda-client.git@ce89e4351eef0f3d2840c713397e013231de25f8
Collecting git+https://github.com/Anaconda-Platform/anaconda-client.git@ce89e4351eef0f3d2840c713397e013231de25f8
  Cloning https://github.com/Anaconda-Platform/anaconda-client.git (to ce89e4351eef0f3d2840c713397e013231de25f8) to /tmp/pip-ms7YuC-build
  Could not find a tag or branch 'ce89e4351eef0f3d2840c713397e013231de25f8', assuming commit.
Collecting clyent>=1.2.0 (from anaconda-client==1.7.2+8.gce89e43)
/home/travis/build/biopython/biopython-wheels/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/home/travis/build/biopython/biopython-wheels/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Could not find a version that satisfies the requirement clyent>=1.2.0 (from anaconda-client==1.7.2+8.gce89e43) (from versions: )
No matching distribution found for clyent>=1.2.0 (from anaconda-client==1.7.2+8.gce89e43)
```

**Comment by peterjc** [comment]
Fixed in 8f0bcde8c7954f777669e3ed90dd6e9a1349d385 - was still using Trusty, and the system's Python 2.7.

Now following Matthew Brett's example and moved to Bionic.
